hw/ip/otbn/doc/_index.md

```diff
@@ -484,48 +503,48 @@ The error that caused the alert can be determined by reading the {{< regref "FAT
 If OTBN was running, this value will also be reflected in the {{< regref "ERR_CODE" >}} register.
 A fatal alert can only be cleared by resetting OTBN through the `rst_ni` line.
 
-# Programmers Guide
 
-<div class="bd-callout bd-callout-warning">
-  <h5>Note</h5>
+# Running applications on OTBN
 
-  This section will be written as we move on in the design and implementation process.
-</div>
+OTBN is a specialized coprocessor which is used from the host CPU.
+This section describes how to interact with OTBN from the host CPU to execute an existing OTBN application.
+The section [Writing OTBN applications]({{< ref "#writing-otbn-applications" >}}) describes how to write such applications.
 
-## Memories
+## High-level operation sequence
 
-The OTBN processor core has access to two dedicated memories:
-an instruction memory (IMEM), and a data memory (DMEM).
-Each memory is 4 kiB in size.
+The high-level sequence by which the host processor should use OTBN is as follows.
 
-The memory layout follows the Harvard architecture.
-Both memories are byte-addressed, with addresses starting at 0.
+1. Write the OTBN application binary to {{< regref "IMEM" >}}, starting at address 0.
+2. Optional: Write constants and input arguments, as mandated by the calling convention of the loaded application,  to {{< regref "DMEM" >}}.
+3. Start the operation on OTBN by writing `1` to {{< regref "CMD.start" >}}.
+   The OTBN application now runs to completion and access to the memories from the host CPU is denied.
+4. Wait for the operation to complete (see below).
+5. Check if the operation was successful by reading the {{< regref "ERR_CODE" >}} register.
+6. Optional: Retrieve results by reading {{< regref "DMEM" >}}, as mandated by the calling convention of the loaded application.
 
-The instruction memory (IMEM) is 32b wide and provides the instruction stream to the OTBN processor;
-it cannot be read or written from user code through load or store instructions.
+OTBN application are run to completion.
+The host CPU can determine if an application has completed either by polling, or by listening to an interrupt.
```

Comment:
Grammar nit: This should probably be `... has completed by either polling {{< regref "STATUS.busy">}} or listening for an interrupt`.

The "either" and "by" need to switch, to make it clear that the application didn't complete by polling. And, since an interrupt is a discrete event, you listen out "for" when it happens. "Listening to" an interrupt means we're listening to the lovely sound it makes over time :-)

Those changes get you to "... has completed by either polling or listening for an interrupt", but then it sounds like you're polling the interrupt, so you also have to say what is being polled.